clarify requestStorageAccess and requestStorageAccessFor (#7725)
GoogleChrome/developer.chrome.com: site/en/docs/privacy-sandbox/related-website-sets-integration/index.md

Embedded resources can use SAA methods to check whether they currently have access to storage, and to request access from the user agent.

> When third-party cookies are blocked but Related Website Sets (RWS) is enabled, Chrome will automatically grant permission in intra-RWS contexts, and will show a prompt to the user otherwise.

Author: perfect thanks

### Checking and requesting storage access

To check whether they currently have access to storage, embedded sites can use [`Document.hasStorageAccess()`](https://developer.mozilla.org/docs/Web/API/Document/requestStorageAccess) method. 

The method returns a promise that resolves with a boolean value indicating whether the document already has access to its cookies or not. The promise also returns true if the iframe is same-origin as the top frame.

{% BrowserCompat 'api.Document.hasStorageAccess' %}

To request access to cookies in a cross-site context embedded sites can use [`Document.requestStorageAccess()`](https://developer.mozilla.org/docs/Web/API/Document/requestStorageAccess) (rSA).

The `requestStorageAccess()` API is meant to be called from within an iframe. That iframe has to have just received user interaction (a [user gesture](https://html.spec.whatwg.org/multipage/interaction.html#user-activation-processing-model), which is required by all browsers), but Chrome additionally requires that at some point in the last 30 days, the user has visited the site that owns that iframe and has interacted with that site specifically—as a top-level document, not in an iframe. 

> `requestStorageAccess()` returns a promise that resolves if the access to storage was granted; however, the promise is rejected, citing the reason, if access was denied due to failure of the interaction reqirement.

Reviewer: ```suggestion
`requestStorageAccess()` returns a promise that resolves if the access to storage was granted; however, the promise is rejected, citing the reason, if access was denied for any reason.
```

There are a few things that might cause the promise to be rejected, not just the user interaction requirement.
